src/cache/raw.rs

```diff
@@ -56,6 +56,17 @@ pub(crate) struct Class {
     pub(crate) members_by_params_offset: u32,
     /// The number of member-by-params entries for this class.
     pub(crate) members_by_params_len: u32,
+    /// Whether this class was synthesized by the compiler.
+    ///
+    /// `0` means `false`, all other values mean `true`.
+    pub(crate) is_synthesized: u32,
+}
+
+impl Class {
+    /// Returns true if this class was synthesized by the compiler.
+    pub(crate) fn is_synthesized(&self) -> bool {
+        self.is_synthesized != false as u32
```

Comment:
Weird way to spell `0` :p, for me just a comparison with 0 would be less confusing. Longterm this could be a bitfield as well.

Reply:
Yeah I wanted it to be very clear that the field is meant to represent a boolean value but I probably overdid it.